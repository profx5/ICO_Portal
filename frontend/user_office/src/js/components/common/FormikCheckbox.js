import React from 'react'
import styled from 'styled-components';
import {media} from 'js/utils/media';

import {Field} from 'formik';
import ErrorMessage from 'js/components/common/ErrorMessage';

import iconCheck from 'img/icons/icon_check.svg';


const FormikCheckbox = ({value, name, labelText, errorStyle, errors, touched, options, icon, handler}) => {

    function handle(handler, event) {
        event.preventDefault();
        handler(1);
    }

    return (
        <Wrapper>
            <InputsWrapper>
                <CheckboxInput component="input" type="checkbox" checked={value} name={name} id={`checkbox-${name}`}/>
                <CheckboxLabel htmlFor={`checkbox-${name}`} className={(errors[name] && touched[name]) ? 'isInvalid' : ''}>
                    {labelText}
                    {icon &&
                        <IconImg onClick={handle.bind(this, handler)} src={icon}/>
                    }
                </CheckboxLabel>
                {errors[name] && touched[name] && <ErrorMessage {...errorStyle} text={errors[name]}/>}
            </InputsWrapper>
        </Wrapper>
    );
}


export default FormikCheckbox;

const Wrapper = styled.div`
    position: relative;
    height: 100%;
`;

const InputsWrapper = styled.div`
    height: 100%;
    display: flex;
    flex-flow: row wrap;
`;

const CheckboxInput = styled(Field)`
    opacity: 0;
    pointer-events: none;
    position: absolute;
    margin-left: 3px;
    margin-top: 3px;
    &:checked + label:before {
        background: url(${iconCheck}) no-repeat center;
    }
`;

const CheckboxLabel = styled.label`
    font-size: 16px;
    letter-spacing: 0.5px;
    color: #0a0a0a;
    margin-right: 20px;
    display: inline-flex;
    align-items: center;
    cursor: pointer;
    ${media.xs} {
        font-size: 12px;
    }
    &:before {
        content: '';
        width: 20px;
        min-width: 20px;
        height: 20px;
        min-height: 20px;
        border: 1px solid #d6dfe6;
        margin-right: 8px;
    }
    &.isInvalid:before {
        border-color: rgb(242,109,109);
    }
`;

const IconImg = styled.img`
    width: 16px;
    height: 16px;
    margin-left: 5px;
`;
