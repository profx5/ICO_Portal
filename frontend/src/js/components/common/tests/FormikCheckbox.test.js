import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FormikCheckbox from 'js/components/common/FormikCheckbox';


configure({ adapter: new Adapter() });

it('should render stateless FormikCheckbox', () => {

    const props = {
        handler: jest.fn(),
        errors: {
            name: 'name'
        }
    }

    const component = shallow(<FormikCheckbox {...props}/>);
    expect(component).toMatchSnapshot();
})
