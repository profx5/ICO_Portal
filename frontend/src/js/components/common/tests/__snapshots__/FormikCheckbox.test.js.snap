// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless FormikCheckbox 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormikCheckbox
    errors={
      Object {
        "name": "name",
      }
    }
    handler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <ForwardRef
          component="input"
          id="checkbox-undefined"
          type="checkbox"
        />
        <ForwardRef
          className=""
          htmlFor="checkbox-undefined"
        />
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            component="input"
            id="checkbox-undefined"
            type="checkbox"
          />,
          <ForwardRef
            className=""
            htmlFor="checkbox-undefined"
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": undefined,
            "className": undefined,
            "component": "input",
            "id": "checkbox-undefined",
            "name": undefined,
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "WrappedComponent": [Function],
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
    opacity: 0;
    pointer-events: none;
    position: absolute;
    margin-left: 3px;
    margin-top: 3px;
    &:checked + label:before {
        background: url(",
                "icon_check.svg",
                ") no-repeat center;
    }
",
              ],
            },
            "displayName": "Styled(FormikConnect(FieldInner))",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
            ],
            "className": "",
            "htmlFor": "checkbox-undefined",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
    font-size: 16px;
    letter-spacing: 0.5px;
    color: #0a0a0a;
    margin-right: 20px;
    display: inline-flex;
    align-items: center;
    cursor: pointer;
    ",
                "@media (max-width: 767px)",
                " {
        font-size: 12px;
    }
    &:before {
        content: '';
        width: 20px;
        min-width: 20px;
        height: 20px;
        min-height: 20px;
        border: 1px solid #d6dfe6;
        margin-right: 8px;
    }
    &.isInvalid:before {
        border-color: rgb(242,109,109);
    }
",
              ],
            },
            "displayName": "styled.label",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "label",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        undefined,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
    height: 100%;
    display: flex;
    flex-flow: row wrap;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
    position: relative;
    height: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef
            component="input"
            id="checkbox-undefined"
            type="checkbox"
          />
          <ForwardRef
            className=""
            htmlFor="checkbox-undefined"
          />
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              component="input"
              id="checkbox-undefined"
              type="checkbox"
            />,
            <ForwardRef
              className=""
              htmlFor="checkbox-undefined"
            />,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "checked": undefined,
              "className": undefined,
              "component": "input",
              "id": "checkbox-undefined",
              "name": undefined,
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "WrappedComponent": [Function],
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
    opacity: 0;
    pointer-events: none;
    position: absolute;
    margin-left: 3px;
    margin-top: 3px;
    &:checked + label:before {
        background: url(",
                  "icon_check.svg",
                  ") no-repeat center;
    }
",
                ],
              },
              "displayName": "Styled(FormikConnect(FieldInner))",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                undefined,
                undefined,
              ],
              "className": "",
              "htmlFor": "checkbox-undefined",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              undefined,
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "
    font-size: 16px;
    letter-spacing: 0.5px;
    color: #0a0a0a;
    margin-right: 20px;
    display: inline-flex;
    align-items: center;
    cursor: pointer;
    ",
                  "@media (max-width: 767px)",
                  " {
        font-size: 12px;
    }
    &:before {
        content: '';
        width: 20px;
        min-width: 20px;
        height: 20px;
        min-height: 20px;
        border: 1px solid #d6dfe6;
        margin-right: 8px;
    }
    &.isInvalid:before {
        border-color: rgb(242,109,109);
    }
",
                ],
              },
              "displayName": "styled.label",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "label",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          undefined,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
    height: 100%;
    display: flex;
    flex-flow: row wrap;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
    position: relative;
    height: 100%;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
